Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -469,6 +474,12 @@ public final class PackageBuilder {
                 if subpath == "" || subpath == "." {
                     return packagePath
                 }
+
+                // Make sure target is not refenced by absolute path
+                if subpath.first == "/" || subpath.first == "~" {
```

Comment:
This is doing similar thing as AbsolutePath's validating [init](https://github.com/apple/swift-package-manager/blob/master/Sources/Basic/Path.swift#L102). I think we should add a similar method for `RelativePath`. Please raise a new PR for landing that method first.

Reply:
Hi, opened [PR](https://github.com/apple/swift-package-manager/pull/1601) for it